xmlrpc/man/update-tcospasswd.1
.TH tcosmd5 1 "Jun 21, 2006" "tcosmd5 man page"

.SH NAME
update-tcospasswd \- TCOS util to generate crypt TCOS passwords files

.SH SYNOPSIS
.B update-tcospasswd
.br

.SH DESCRIPTION

This manpage show options for 
.B update-tcospasswd
script.

.PP
\fBupdate-tcospasswd\fP is a small shell script that create TCOS crypt 
password files with some questions.


.SH BUGS
Contact Mario Izquierdo <mariodebian@gmail.com>

Info about TCOS project: http://soleup.eup.uva.es/mediawiki/index.php/TCOS

.SH SEE ALSO
tcos.conf(5)

.SH AUTHOR
Mario Izquierdo (mariodebian) <mariodebian@gmail.com>


Changelog
tcosmonitor (0.1.0) unstable; urgency=low

  * Not build LTSP or PXES packages anymore (not clean, not debian)
  * Not get busybox-static
  * Use python-support to bytecompile packages
  * Update debhelper Build-dep
  * Move TcosMonitor desktop icon to Admin menu
  * Hide tcospersonalize menu
  * Not ignore debhelper in tcos-tcosmonitor.postinst
  * Change tcospersonalize config dir to new /var/lib/tcos/tftp/config

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 18 Feb 2007 13:36:53 +0100

tcosmonitor (0.0.19.1) unstable; urgency=low

  * xmlrpc/*:
     - Change #ifdef DEBUG for a new function dbgtcos(); read TCOS_DEBUG env var

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Thu,  8 Feb 2007 18:33:04 +0100

tcosmonitor (0.0.18.2) unstable; urgency=low

  * Downgrade initramfs-tools-tcos dependencies to >= 0.59
    tcosmonitor must be installed before initramfs-tools-tcos 0.60

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun,  4 Feb 2007 22:04:25 +0100

tcosmonitor (0.0.18.1) unstable; urgency=low

  * hooks-addons/tcosmonitor:
     - remove copy od seq (fixed in initramfs-tools-tcos 0.60)
  * debian/control:
     - update version depends
  * Remove hex2ascii bin => initramfs-tools-tcos

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat,  3 Feb 2007 20:39:01 +0100

tcosmonitor (0.0.17.1) unstable; urgency=low

  * xmlrpc/*.c:
     - Remove tcosxmlrpc verbose messages
  * debian/control, hooks-addons/tcosmonitor:
     - Add /usr/bin/eject and eject cdrom from TcosDevices (daemon)
  * tcos-devices.py:
     - Remove sys.exit(1) from tcos-devices (retry cookie don't match)
  * xmlrpc/sh/devicesctl.sh
     - Add --eject method

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 29 Jan 2007 16:13:43 +0100

tcosmonitor (0.0.16.2) unstable; urgency=low

  * Change /bin/sh to /bin/bash (solve ubuntu dash problems)

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 29 Jan 2007 16:02:30 +0100

tcosmonitor (0.0.16.1) unstable; urgency=low

  * New stable release.
  * Add vertical label in all screenshots.
  * Add uptime var (readed in TCOS general info)
  * Change preferences minicrot_size default (20) and step (1)
  * Change version string in shared.py (in future read from debian/changelog when building package)
  * Fix typo in tcos-server-utils.py

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 26 Dec 2006 23:35:11 +0100

tcosmonitor (0.0.15.14) unstable; urgency=low

  * Add New toolbar button (All Hosts) that show all_actions_menu
  * Add new global action: "Capture all hosts screenshots"
  * New Preferences var: miniscrot_size (default 15%)
  * tcos-volume-manager:
     - add refresh button to reload all sliders values.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 23 Dec 2006 21:10:31 +0100

tcosmonitor (0.0.15.13) unstable; urgency=low

  * Fix typo on on_another_screenshot() Worker => Workers

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 11 Dec 2006 16:12:48 +0100

tcosmonitor (0.0.15.12) unstable; urgency=low

  * Add set_sensitive() to TreeView before/after updating list or 
    connecting to thin client.
  * Fix app/msg dialog, print hostname/username in new lines.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon,  4 Dec 2006 09:46:20 +0100

tcosmonitor (0.0.15.11) unstable; urgency=low

  * Fix LocalData.py who |grep "%s "  (add space after %s)

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 29 Nov 2006 19:07:45 +0100

tcosmonitor (0.0.15.10) unstable; urgency=low

  * Put 1d if hours connected < 0 (connected yesterday)

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 29 Nov 2006 18:46:20 +0100

tcosmonitor (0.0.15.9) unstable; urgency=low

  * Fix minor problem in import headers in shared.py file.
  * Fix version retrieve error (disable split in mayor, med and minor)
  * Update web and translation.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 28 Nov 2006 21:36:10 +0100

tcosmonitor (0.0.15.8) unstable; urgency=low

  * xmlrpc/Makefile
          -DVERSION=\"fooo\" not use "'foo'"
  * LocalData.py
       Add print_debug in version get
  * shared.py Add some hidden process to list
       ( xauth, tcos-devices, tcos-volume-manager, notification-daemon, bash )

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 19 Nov 2006 20:14:59 +0100

tcosmonitor (0.0.15.7) unstable; urgency=low

  * allow to enable/disable tcos-devices and tcos-volume-manager
    from a new conf file: /etc/tcos/tcosmonitor.conf in runtime.

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed,  8 Nov 2006 13:11:41 +0100

tcosmonitor (0.0.15.6) unstable; urgency=low

  * tcos-volume-manager.py, tcos-volume-manager.glade, tcos-volume-manager.desktop
   - iconize in systray tcos-volume-manager
   - new systray icon for tcos-volume-manager
   - add xdg/autostart tcos-volume-manager (desktop file)
   - exit if xauth dont match
   - mainwindow hide by default in glade file

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 28 Oct 2006 15:38:06 +0200

tcosmonitor (0.0.15.5) unstable; urgency=low

  * tcos-devices.py
    - fix get_value(), DEVICE is not a udev var, varname is DEVNAME
    - fix get_local_mountpoint() fslabel and fsvendor

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 28 Oct 2006 11:54:13 +0200

tcosmonitor (0.0.15.4) unstable; urgency=low

  * tcos-devices.py, udev/tcos-udevd.sh , udev/050_tcos_devices.rules
    - Change udev to control all SUBSYSTEM=="block" events
    - Delete duplicate events, kernel do 4 umount and 1 mount event for mounting a floppy

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 25 Oct 2006 17:24:14 +0200

tcosmonitor (0.0.15.3) unstable; urgency=low

  * TcosPersonalize: run as root only
  * TcosActions: fix update info bug
  * TcosXmlRpc, TcosXauth, TcosConf, ping: import gtk modules if env var DISPLAY is set


xmlrpc/sh/update-tcospasswd.sh
#!/bin/bash

# update-tcospasswd.sh shell script to generate tcospasswd
#                      to PXES, LTSP or TCOS
#  2006-09-09 14:22:40 mariodebian $
#
# This file is part of tcosxmlrpc.
#
# tcosxmlrpc is free software; you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# tcosxmlrpc is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with tcosxmlrpc; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
# USA.

if [ "$(whoami)" != "root" ]; then
  echo "Error: Need to exec as root user."
  exit 1
fi

_is_ltsp=$(echo /opt/ltsp*)
_is_pxes=$(echo /opt/pxes*)
_is_tcos=$(echo /etc/tcos)

if [ -d ${_is_ltsp} ]; then
  ltsp=1
fi
if [ -d ${_is_pxes} ]; then
  pxes=1
fi
if [ -d ${_is_tcos} ]; then
  tcos=1
fi


make_passwd(){
$(which tcospasswd)
}


echo "Arquitecture to update password:"
echo ""
if [ ${ltsp} ]; then
echo "  * LTSP"
fi

if [ ${pxes} ]; then
echo "  * PXES"
fi

if [ ${tcos} ]; then
echo "  * TCOS"
fi

echo -n "              : "
read _arq

if [ "$_arq" = "LTSP" ]; then
  fpass=/opt/ltsp/i386/etc/tcospasswd

elif [ "$_arq" = "PXES" ]; then
  fpass=$(echo /opt/pxes*)/stock/dist/etc/tcospasswd

elif [ "$_arq" = "TCOS" ]; then
  fpass=/etc/tcos/tcospasswd
else
  echo "Error: Arquitecture: ${_arq} not supported, please use LTSP, PXES, or TCOS"
  exit 1
fi

echo -n "Username: "
read _username

string="$_username:$(make_passwd)"


if [ ! -d $(dirname ${fpass}) ]; then 
  echo "Error: $(dirname ${fpass}) don't exists."
  exit 1
fi

echo -n "I will write this file:
  ${fpass}
          Continue? [Y/n] "
read _ans

if [ "${_ans}" != "n" ]; then
  echo ${string} > ${fpass}
  echo ""
  echo "In case of PXES or TCOS, please update the"
  echo "boot image with pxesconfig or gentcos."
  echo "Done."
  exit 0
fi

echo "Aborted."
exit 1

